light/rpc/client.go

```diff
@@ -462,6 +463,18 @@ func (c *Client) DataCommitment(
 	return c.next.DataCommitment(ctx, beginBlock, endBlock)
 }
 
+// DataRootInclusionProof calls rpcclient#DataRootInclusionProof method and returns
+// a merkle proof for the data root of block height `height` to the set of blocks
+// defined by `beginBlock` and `endBlock`.
+func (c *Client) DataRootInclusionProof(
+	ctx context.Context,
+	height uint64,
+	beginBlock uint64,
+	endBlock uint64,
+) (*ctypes.ResultDataRootInclusionProof, error) {
+	return c.next.DataRootInclusionProof(ctx, height, beginBlock, endBlock)
```

Comment:
Currently, the light client is relying on whatever the full node responds with. Is there any way that the light client can verify this response themselves. Could they not fetch all the headers for the range of heights and build the inclusion proof themselves?

Reply:
that's true, do we want to support light clients generating the proofs ? @evan-forbes what do you think please?

Reply:
yeah that makes a lot of sense, and is something we can plan for in celestia-node. Until then, nodes that need to produce proofs for the QGB will have to have access to a trusted celestia full node :grimacing:

Reply:
https://github.com/celestiaorg/celestia-core/issues/924